src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.MacCatalyst.cs

```diff
@@ -7,7 +7,16 @@ public static partial class Environment
     {
         private static OperatingSystem GetOSVersion()
         {
-            Version version = new Version(Interop.Sys.iOSSupportVersion());
+            Version version = Version.Parse(Interop.Sys.iOSSupportVersion());
+
+            // Check if build and revision are -1 and default them to 0
+            int major = version.Major;
+            int minor = version.Minor;
+            int build = version.Build < 0 ? 0 : version.Build;
+            int revision = version.Revision < 0 ? 0 : version.Revision;
```

Comment:
This change will also affect value of  `Environment.OSVersion.Version.ToString()` and make it different from the natural OS version: e.g. `18.0` vs. `18.0.0.0`. This would require filling a breaking change notice.

I think it may be better to just fix the comparison as @am11 suggested. It is less breaking change and keeps the version formatted like the natural OS version.